src/saltext/vmware/utils/esxi.py

```diff
@@ -155,3 +159,167 @@ def list_hosts(service_instance):
+    if host.summary.runtime.connectionState == "disconnected":
+        return host.summary.runtime.connectionState
+    task = host.DisconnectHost_Task()
+    host = utils_common.wait_for_task(task, name, "disconnect host task")
+    return host.summary.runtime.connectionState
+
+
+def reconnect_host(name, service_instance):
+    """
+    Reconnects host from vCenter instance
+
+    Returns connection state of host
+
+    name
+        Name of host.
+
+    service_instance
+        The Service Instance Object from which to obtain host.
+    """
+    host = utils_common.get_mor_by_property(service_instance, vim.HostSystem, name)
+    if host.summary.runtime.connectionState == "connected":
+        return host.summary.runtime.connectionState
+    task = host.ReconnectHost_Task()
+    ret_host = utils_common.wait_for_task(task, name, "reconnect host task")
+    return ret_host.summary.runtime.connectionState
+
+
+def move_host(name, cluster_name, service_instance):
+    """
+    Move host to a different cluster.
+
+    Returns connection state of host
+
+    name
+        Name of host.
+
+    cluster_name
+        Name of cluster to move host to.
+
+    service_instance
+        The Service Instance Object from which to obtain host.
+    """
+    host_ref = utils_common.get_mor_by_property(service_instance, vim.HostSystem, name)
+    cluster_ref = utils_common.get_mor_by_property(
+        service_instance, vim.ClusterComputeResource, cluster_name
+    )
+    host_dc = utils_common.get_mors_type(host_ref, vim.Datacenter)
+    host_cluster = utils_common.get_mors_type(host_ref, vim.ClusterComputeResource)
+    cluster_dc = utils_common.get_mors_type(cluster_ref, vim.Datacenter)
+    if host_dc != cluster_dc:
+        raise salt.exceptions.VMwareApiError("Cluster has to be in the same datacenter")
+    task = cluster_ref.MoveInto_Task([host_ref])
+    utils_common.wait_for_task(task, cluster_name, "move host task")
+    return f"moved {name} from {host_cluster.name} to {cluster_ref.name}"
+
+
+def remove_host(name, service_instance):
+    """
+    Removes host from vCenter instance.
+
+    Returns connection state of host
+
+    name
+        Name of host.
+
+    service_instance
+        The Service Instance Object from which to obtain host.
+    """
+    host = utils_common.get_mor_by_property(service_instance, vim.HostSystem, name)
+    task = host.Destroy_Task()
+    utils_common.wait_for_task(task, name, "destroy host task")
+    return f"removed host {name}"
+
+
+def _format_ssl_thumbprint(number):
+    """
+    Formats ssl cert number
+
+    number
+        Number to be formatted into ssl thumbprint
+    """
+    string = str(number)
+    return ":".join(a + b for a, b in zip(string[::2], string[1::2]))
+
+
+def _get_host_thumbprint(ip):
+    """
+    Returns host's ssl thumbprint.
+
+    ip
+        IP address of host.
+    """
+
+    ctx = ssl.create_default_context(ssl.Purpose.CLIENT_AUTH)
```

Comment:
I'm sorry for saying this now.

It's common to have unverified certs in dev, we should probably pass a flag to `add_host` that can disable it, maybe `verify_certificate` defaulting to true?